File: answer.java
import java.io.*;
import java.util.*;

public class answer
{
   public answer(){}
   
   public static void main(String args[])
   {
      
      try
      {
		 // Create the BufferedReader to read in the file input.
         BufferedReader in = new BufferedReader(new FileReader("answer.in"));
         
		 // Get the number of data sets we need to process
         int numProducts = Integer.parseInt(in.readLine());
         
         for(int i = 0; i < numProducts; i++)
         {
		    // Get the two numbers we need by reading the line in and passing it
			// to the StringTokenizer.
            StringTokenizer st = new StringTokenizer(in.readLine());
            int firstNumber = Integer.parseInt(st.nextToken());
            int secondNumber = Integer.parseInt(st.nextToken());
            
			// See if the numbers are 6 or 9, and if so, we need to print
			// out 42.
            if((firstNumber == 6 && secondNumber == 9) ||
               (firstNumber == 9 && secondNumber == 6))
            {
               System.out.println(42);
            }
		    // Otherwise, just print out the product.
            else
            {
               System.out.println(firstNumber*secondNumber);
            }
         }
      }catch(Exception e){
         e.printStackTrace();
      }
   }
}
      

File: answer.c
#include <stdio.h>

int main()
{
   FILE   *infile;
   char   line[80];
   int    numSets;
   int    i;
   int    num1;
   int    num2;

   /* Open the input file */
   infile = fopen("answer.in", "r");

   /* Get the first line and get the number of tests from it */
   fgets(line, sizeof(line), infile);
   sscanf(line, "%d", &numSets);

   /* Loop through the sets */
   for (i=0; i < numSets; i++)
   {
      /* Get the two numbers */
      fgets(line, sizeof(line), infile);
      sscanf(line, "%d %d", &num1, &num2);

      /* If they are 6 and 9, output the true Answer (otherwise, the product) */
      if ( ((num1 == 6) && (num2 == 9)) ||
           ((num1 == 9) && (num2 == 6)) )
         printf("42\n");
      else
         printf("%d\n", num1 * num2);
   }

   /* Exiting fine */
   return 0;
}

